### PR TITLE
Ch auction backend setup

### PR DESCRIPTION
**Pull request by otitoju**
#### What does this PR do?
-Enable the MVC folder and created environment for all auction APIS
#### Description of Task to be completed?
-postgres and sequelize setup
-functions for the controllers
-auction models
-auction APIS
#### How should this be manually tested?
-clone the repo
-cd inside auction-backend
-run npm install
-npm start

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules
+dist
```
Comment (codinger41) [inline .gitignore]: Add a new line here.

File: server/.babelrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "presets": ["es2015", "stage-2"]
+}
```
Comment (codinger41) [inline server/.babelrc]: Add a newline here.

File: server/app.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+const port = 4000
+
+app.get('/', (req, res) => {
+    res.send(`<h1>Going Remote 2020 simulation server...`)
+})
+
+// server listen
+app.listen(port, () => {
+    console.log(`Going Remote server started ${port}`)
+})
```
Comment (codinger41) [inline server/app.js]: Add a newline here.

File: server/bin/dev [hunk 4]
```diff
@@ -0,0 +1,2 @@
+require('babel-register')
+require('./../app')
```
Comment (codinger41) [inline server/bin/dev]: Add a newline here.

File: server/bin/prod [hunk 5]
```diff
@@ -0,0 +1 @@
+require('./../../dist/app')
```
Comment (codinger41) [inline server/bin/prod]: Add a newline here.

**Comment by codinger41** [comment]
The reason we can't see the controllers and routes is that the controller and routes folders are empty. Add a dummy route and a dummy controller so we can see these files. Good job @otitoju

File: server/controller/usercontroller.js [hunk 6]
```diff
@@ -0,0 +1,2 @@
+import User from '../model/user'
+
```
Comment (codinger41) [inline server/controller/usercontroller.js]: @otitoju  let's add a dummy controller to return a dummy response message.

File: server/routes/routes.js [hunk 7]
```diff
@@ -0,0 +1,2 @@
+//all auction routes here
+
```
Comment (codinger41) [inline server/routes/routes.js]: Let's also make a route here then link it with the dummy controller.

File: server/model/user.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+const User = []
+
+export default User;
```
Comment (codinger41) [inline server/model/user.js]: Let's add a newline at the end of this file.

File: server/controller/usercontroller.js (hunk 6 above)
Comment (otitoju) [reply server/controller/usercontroller.js]: Alright

File: server/routes/routes.js (hunk 7 above)
Comment (otitoju) [reply server/routes/routes.js]: Alright

File: server/controller/usercontroller.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
+var data = [
+    {id: 1, name: 'John Doe', email: 'johndoe@gmail.com', password: '12345'},
+    {id: 2, name: 'Janet Doe', email: 'janetdoe@gmail.com', password: '123456'},
+    {id: 3, name: 'Sam Doe', email: 'samdoe@gmail.com', password: '123457'},
+]
+
+exports.getAllUser = async (req, res) => {
```
Comment (codinger41) [inline server/controller/usercontroller.js]: Let's pay attention to English here, this should be getAllUsers

File: server/controller/usercontroller.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    try {
+        const info = await data.findAll({})
```
Comment (codinger41) [inline server/controller/usercontroller.js]: How did we come about the .findAll() method?

File: server/controller/usercontroller.js [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+        })
+    } catch (e) {
+        return res.status(500).json({error: e.message})
+    }
+}
+
+exports.getSingleUser = async (req, res) => {
+    try {
+        const { id } = req.params
+        const info = await data.findOne({ where: {id: id }})
```
Comment (codinger41) [inline server/controller/usercontroller.js]: Let's work with what we have now. What we have is an array and not an actual model.

File: server/routes/routes.js [hunk 12]
```diff
@@ -0,0 +1,10 @@
+import { Router } from 'express'
+const router = Router()
+import { catchErrors } from '../handlers/errorHandler'
+import usercontroller from '../controller/usercontroller'
+
+router.get('/api/users', catchErrors(usercontroller.getAllUser))
+router.get('/api/user/:id', catchErrors(usercontroller.getSingleUser))
+router.delete('/api/user/del/:id', catchErrors(usercontroller.deleteUser))
+
+export default router;
```
Comment (codinger41) [inline server/routes/routes.js]: Add a newline here

**Pull request merged**